### DISCUSSION TITLE
How to open menu links in a new tab?

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [zauberzeug/nicegui#1228] [Q&A]
### Question

The nicescad project is making progress. Version 0.0.3 has a demo at http://nicescad.bitplan.com/

there are some follow up issues i'd like to get fixed:

**Menu Links**
```python
def link_button(self, name: str, target: str, icon_name: str):
        """
        Creates a button with a specified icon that opens a target URL upon being clicked.
    
        Args:
            name (str): The name to be displayed on the button.
            target (str): The target URL that should be opened when the button is clicked.
            icon_name (str): The name of the icon to be displayed on the button.
    
        Returns:
            The button object.
        """
        with ui.button(name,icon=icon_name) as button:
            button.on("click",lambda: (ui.open(target)))
        return button
```
The menu links should partly open in a new tab and according to ui.open the socket needs to be taken into account. How would i add those two features?

**Comment by WolfgangFahl** [comment]
@Smug246  combining button and link would indeed be great. Doing this on click might just show the link but not click it. The link_button function is indeed trying to do a work-around.

**Comment by rodja** [comment]
As @falkoschindler said in #960 

> `ui.open` should accept a new_tab argument like `ui.link` does

@WolfgangFahl can you create a pull request?

**Reply by WolfgangFahl** [reply]
Falko was quicker with #960  but  i would have been stuck here anyways:
```python
def open(self, target: Union[Callable[..., Any], str]) -> None:
        """Open a new page in the client."""
        path = target if isinstance(target, str) else globals.page_routes[target]
        outbox.enqueue_message('open', path, self.id)
```
Because i don't know the message format details.

**Reply by WolfgangFahl** [reply]
I'll wait for the PR to be accepted
